internal/dkg/state_machine.go

```diff
@@ -740,10 +743,19 @@ func validateForAllDKGs(currentState *DBState, terms *drand.ProposalTerms) error
 		return ErrTimeoutReached
 	}
 
-	if int(terms.Threshold) > len(terms.Joining)+len(terms.Remaining) {
+	nodeCount := len(terms.Joining) + len(terms.Remaining)
+	if int(terms.Threshold) > nodeCount {
 		return ErrThresholdHigherThanNodeCount
 	}
 
+	if nodeCount < int(currentState.Threshold) {
```

Comment:
Shouldn't that be rather
```suggestion
	if len(terms.Remaining) < int(currentState.Threshold) {
```

Reply:
I changed this but github doesn't show outdated, not sure why

Reply:
I think my comment still holds:

if there aren't enough remaining nodes (+ leader) in the proposal, we cannot expect to run a successful DKG since we need at least threshold nodes with a secret share to do a resharing (and since currently the leavers aren't participating in the resharing IIRC the code)

Reply:
I think you only changed it below in `validateReshareTerms`

Reply:
ah oops, my find/replace was on regex mode fuuu